### PR TITLE
[dtest] Update hipMemset test

### PR DESCRIPTION
**Pull request by gargrahul**
Added result check and hipMemsetAsync test

### CODE REVIEW

File: tests/src/runtimeApi/memory/hipMemset.cpp [hunk 1]
```diff
@@ -37,32 +37,66 @@ THE SOFTWARE.
...
+    char *A_h;
+    bool testResult = true;
 
-    HIPCHECK(hipMalloc(&A_d, Nbytes));
+    HIPCHECK ( hipMalloc((void**)&A_d, Nbytes) );
     A_h = (char*)malloc(Nbytes);
+    hipStream_t stream;
+    HIPCHECK(hipStreamCreate(&stream));
+    HIPCHECK ( hipMemsetAsync(A_d, memsetval, Nbytes, stream ));
+    HIPCHECK ( hipMemcpy(A_h, (void*)A_d, Nbytes, hipMemcpyDeviceToHost));
```
Comment (mangupta) [inline tests/src/runtimeApi/memory/hipMemset.cpp]: Probably need a streamsync before this.

File: tests/src/runtimeApi/memory/hipMemset.cpp [hunk 2]
```diff
@@ -37,32 +37,66 @@ THE SOFTWARE.
...
+    return testResult;
+}
 
-    passed();
+int main(int argc, char *argv[])
+{
+    HipTest::parseStandardArguments(argc, argv, true);
+    bool testResult = false;
+    HIPCHECK(hipSetDevice(p_gpuDevice));
+    testResult = testhipMemset(memsetval, p_gpuDevice);
```
Comment (mangupta) [inline tests/src/runtimeApi/memory/hipMemset.cpp]: This block can be rewritten as:
```
bool testResult = true;
testResult &= testhipMemset(memsetval, p_gpuDevice);
testResult &= testhipMemsetAsync(memsetval, p_gpuDevice);
if (testResult) passed();
failed();
```

File: tests/src/runtimeApi/memory/hipMemset.cpp [hunk 3]
```diff
@@ -37,32 +37,66 @@ THE SOFTWARE.
...
-            failed("mismatch at index:%d computed:%02x, memsetval:%02x\n", i, (int)A_h[i],
-                   (int)memsetval);
+            testResult = false;
+            printf("mismatch at index:%d computed:%02x, memsetval:%02x\n", i, (int)A_h[i], (int)memsetval);
+            break;
         }
     }
+    hipFree((void*)A_d);
+    free(A_h);
+    return testResult;
```
Comment (mangupta) [inline tests/src/runtimeApi/memory/hipMemset.cpp]: Missing a stream destroy here.

**Comment by gargrahul** [comment]
Thanks. Fixed the test.

**Pull request merged**